src/qt/diagnosticsdialog.cpp

```diff
@@ -0,0 +1,375 @@
+#include "diagnosticsdialog.h"
+#include "ui_diagnosticsdialog.h"
+
+DiagnosticsDialog::DiagnosticsDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::DiagnosticsDialog)
+{
+    ui->setupUi(this);
+
+    networkManager = new QNetworkAccessManager(this);
+    connect(networkManager, SIGNAL(finished(QNetworkReply *)), this, SLOT(getGithubVersionFinished(QNetworkReply *)));
+}
+
+DiagnosticsDialog::~DiagnosticsDialog()
+{
+    delete ui;
+}
+
+void DiagnosticsDialog::GetData() {
+    std::string cpiddata = GetListOf("beacon");
+    std::string sWhitelist = GetListOf("project");
+    int64_t superblock_age = GetAdjustedTime() - mvApplicationCacheTimestamp["superblock;magnitudes"];
+    double popularity = 0;
+    std::string consensus_hash = GetNeuralNetworkSupermajorityHash(popularity);
+    std::string sAge = ToString(superblock_age);
+    std::string sBlock = mvApplicationCache["superblock;block_number"];
+    std::string sTimestamp = TimestampToHRDate(mvApplicationCacheTimestamp["superblock;magnitudes"]);
+    printf("Pushing diagnostic data...");
+    double lastblockage = PreviousBlockAge();
+    double PORDiff = GetDifficulty(GetLastBlockIndex(pindexBest, true));
+    syncData = "<WHITELIST>" + sWhitelist + "</WHITELIST><CPIDDATA>"
+        + cpiddata + "</CPIDDATA><QUORUMDATA><AGE>" + sAge + "</AGE><HASH>" + consensus_hash + "</HASH><BLOCKNUMBER>" + sBlock + "</BLOCKNUMBER><TIMESTAMP>"
+        + sTimestamp + "</TIMESTAMP><PRIMARYCPID>" + KeyValue("cpid") + "</PRIMARYCPID><LASTBLOCKAGE>" + ToString(lastblockage) + "</LASTBLOCKAGE><DIFFICULTY>" + RoundToString(PORDiff,2) + "</DIFFICULTY></QUORUMDATA>";
+    testnet_flag = fTestNet ? "TESTNET" : "MAINNET";
+}
+
+int DiagnosticsDialog::VarifyBoincPath() {
+    boost::filesystem::path boincPath = (boost::filesystem::path) GetBoincDataDir();
```

Comment:
boost::filesystem::path boincPath = GetBoincDataDir();

this would be consistent with existing code? @denravonska

Reply:
GetBoincDataDir() returns a std::string not a boost path, that's all the cast if there for. I could probably strip out the boost if that would be the prefered way of doing this?

Reply:
yea ur actually right about that i forgot we GetDataDir was done with boost. however it was my understanding that path converts as we do use strings in ours for wallet backups. i''ll have to look this one up :)

Reply:
It is constructible from std::string, right? If so, `boost::filesystem::path boincPath(GetBoincDataDir());`